src/logic/BrazeMessages.tsx

```diff
@@ -178,26 +153,26 @@ class BrazeMessages implements BrazeMessagesInterface {
 
         // Otherwise we'll wait for a fresh message to arrive, returning the
         // data from the cache (where it will have already been added)
-        return this.freshEndOfArticleMessage.then(() => {
-            const freshMessageFromCache = this.cache.peek(
-                'EndOfArticle',
-                this.appboy,
-                this.errorHandler,
+        return this.freshBannerMessage.then(() => {
```

Comment:
I think because we're in the generic `getMessageForSlot` method here that this should also be generic (either `freshBannerMessage` or `freshEndOfArticleMessage` depending on the `slotName`). I'm surprised that this didn't cause a failing test. Would be good to fix this by writing a suitable failing test first so we've got that confidence going forward.